File: api-ruby/samples/server/widget.rb
require 'sinatra/base'
require_relative 'config'
module WidgetRoutes
  def self.registered(app)
    app.get '/widget' do
      send_file File.join(settings.public_folder, 'widget.html')
    end

    app.post '/create-widget-session' do
      redirectUrl = params[:redirectUrl]
      verificationProfileId = ENV['TRINSIC_VERIFICATION_PROFILE_ID']
      req = TrinsicApi::CreateWidgetSessionRequest.new({
        redirect_url: redirectUrl,
        verification_profile_id: verificationProfileId
      })

      result = TrinsicServices::SESSIONS.create_widget_session({
        create_widget_session_request: req
      })
      json result.to_body
    end

  end
end

File: api-typescript/samples/server/src/widget.ts
import { CreateWidgetSessionRequest, SessionsApi } from "@trinsic/api";
import path from "path";
import { Express } from "express"

export function widgetRoutes(app: Express, sessionsApi: SessionsApi) {
  app.get("/widget", (req: any, res: any) => {
    res.sendFile(
      path.join(__dirname, "../../../../ui-web/samples/dist/widget.html")
    );
  });

  app.post("/create-session", async (req: any, res: any) => {
    const redirectUrl = req.query.redirectUrl;
    const result = await sessionsApi.createWidgetSession({ redirectUrl:  redirectUrl });
    console.debug("Created session", result);
    res.send(result);
  });
}